lib/common/output_text.c

```diff
@@ -148,7 +148,13 @@ text_output_xml(pcmk__output_t *out, const char *name, const char *buf) {
     private_data_t *priv = out->priv;
 
     CRM_ASSERT(priv != NULL);
-    pcmk__indented_printf(out, "%s", buf);
+
+    if (name != NULL) {
+        pcmk__indented_printf(out, "%s:\n", name);
+        fprintf(out->dest, "%s\n", buf);
+    } else {
+        pcmk__indented_printf(out, "%s", buf);
+    }
```

Comment:
That seems like a reasonable approach.  I'll drop this patch.

Reply:
crm_resource can't do this because outputting the XML happens way down in the various resource functions.  However, I think I can do something in those instead.